client/src/rr/dnssec/keypair.rs

```diff
@@ -495,6 +495,34 @@ impl KeyPair {
     }
 }
 
+impl Clone for KeyPair {
+    /// *Note* clone with PKey and OpenSSL is not cheap.
```

Comment:
The *ring*-based build fails because `Ed25519KeyPair` isn't `Clone`. I should probably make it `Clone`. OTOH, I don't think this PR should require cloning a `KeyPair`. I suggest instead that the things that own a `KeyPair` should change to be hold a `Rc<XXX>` for the highest-level type of `XXX` there is, probably `Signer`.

In particular, this all looks too error-prone as written. A type like `KeyPair` should be `#[derive(Clone)]` or not `Clone` at all. If the things it contains aren't `Clone` then it's better to fix that than to try to work around it.

Reply:
Yes. Good feedback. I'll look at making this sync.

Reply:
Actually, if the goal of the PR is to just make things `Send`, then it might be helpful to know that *ring* types are all `Send` already (otherwise, it's a bug). At least when *ring* is being used, you shouldn't need this type to implement `Clone` just to make other things `Send`.

Now, I see the commit message is trying to make these types `Clone and Send`. IMO, it's a bad idea to make them `Clone` without a specific use case.